src/System.Private.CoreLib/src/System/Threading/ClrThreadPool.WorkerThread.cs

```diff
@@ -22,11 +22,11 @@ private static class WorkerThread
             
             private static void WorkerThreadStart()
             {
-                // TODO: Event: Worker Thread Start event
+                ClrThreadPoolEventSource.Log.WorkerThreadStart(ThreadCounts.VolatileReadCounts(ref ThreadPoolInstance._separated.counts).numExistingThreads);
                 RuntimeThread currentThread = RuntimeThread.CurrentThread;
                 while (true)
                 {
-                    // TODO: Event:  Worker thread wait event
+                    ClrThreadPoolEventSource.Log.WorkerThreadWait(ThreadCounts.VolatileReadCounts(ref ThreadPoolInstance._separated.counts).numExistingThreads);
```

Comment:
Might be a bit cleaner to make it an infinite loop, move the loop condition and break out, and put the event call at the top of the loop, but this is fine too.